desc/interpolate.py

```diff
@@ -8,6 +8,100 @@
 from desc.backend import jnp
 
 
+def fft_interp1d(f, n, sx=None, dx=1):
+    """Interpolation of a 1d periodic function via FFT.
+
+    Parameters
+    ----------
+    f : ndarray, shape(nx, ...)
+        Source data. Assumed to cover 1 full period, excluding the endpoint.
+    n : int
+        Number of desired interpolation points.
+    sx : ndarray or None
+        Shift in x to evaluate at. If original data is f(x), interpolates to f(x + sx)
+    dx : float
+        Spacing of source points
+
+    Returns
+    -------
+    fi : ndarray, shape(n, ..., len(sx))
+        Interpolated (and possibly shifted) data points
+    """
+    c = jnp.fft.ifft(f, axis=0)
```

Comment:
kind of confused, I thought that the interp FFT sohuld work by FFT the input f, then eventually calling ifft to evaluate the coefficients of f at more grid points? I must be misssing something because this starts with ifft?

Reply:
They're equivalent (`fft` and `ifft` are transposes of each other), it just depends on the sign convention (see https://numpy.org/doc/stable/reference/routines.fft.html#implementation-details)

I'm more used to the positive sign on the forward transform, which in numpy/jax is `ifft`